### DISCUSSION TITLE
how to add custom functionality manually using golang?

### DISCUSSION THREAD

**Discussion by saatvik333** [cosmos/community#31] [Chain Builders (Cosmos SDK)] [answered]
So I have been using Ignite to build a chain for my purpose, which requires adding additional functionality.
I want to build a chain using cosmos SDK that should be able to interact with other chains to receive data and verify the batches.

Any help is greatly appreciated (:

**Comment by facundomedica** [comment, marked as answer]
> So I have been using Ignite to build a chain for my purpose, which requires adding additional functionality.

What you describe is what we call a "module". A module can have its own messages and store, it can also query and execute messages in other modules in your chain and it can execute actions at the beginning and end of every block. Please take a look at these docs: https://docs.cosmos.network/main/building-modules/intro

> I want to build a chain using cosmos SDK that should be able to interact with other chains to receive data and verify the batches.

Regarding your specific use case I can't give you very specific advice, other than to investigate about oracles or interchain queries.
